src/Microsoft.TestPlatform.CrossPlatEngine/Discovery/DiscovererEnumerator.cs

```diff
@@ -277,6 +304,49 @@ private void SetAdapterLoggingSettings(IMessageLogger messageLogger, IRunSetting
             return result;
         }
 
+        /// <summary>
+        /// Get assembly type to sources map.
+        /// </summary>
+        /// <param name="sources">Sources.</param>
+        /// <param name="assemblyType">Assembly type.</param>
+        /// <returns>Sources with mathcing assembly type.</returns>
+        private static IDictionary<AssemblyType, IEnumerable<string>> GetAssemblyTypeToSoucesMap(IEnumerable<string> sources, IAssemblyProperties assemblyProperties)
+        {
+            var assemblyTypeToSoucesMap = new Dictionary<AssemblyType, IEnumerable<string>>()
+            {
+                { AssemblyType.Native, new List<string>()},
+                { AssemblyType.Managed, new List<string>()},
+                { AssemblyType.None, new List<string>()}
+            };
+
+            if (sources != null && sources.Any())
+            {
+                foreach (string source in sources)
+                {
+                    var sourcesList = IsAssembly(source) ?
+                        assemblyTypeToSoucesMap[assemblyProperties.GetAssemblyType(source)] :
+                        assemblyTypeToSoucesMap[AssemblyType.None];
+
+                    ((List<string>)sourcesList).Add(source);
```

Comment:
Can we use `IList` here?

Reply:
IList doesn't take a string, & does not have Add method, would need to create  another IList for source.

Reply:
done